### PR TITLE
Update for java11 readiness

### PR DESCRIPTION
**Pull request by batmat**
The core goal here was to fix a failure detected on Java 11 through PCT runs.

Had to bump a few things around to get it running. 

This should also fix #33 though it's hard to be sure, given it's a bit of a chicken-and-egg issue.

### CODE REVIEW

File: src/test/java/org/jenkinsci/plugins/tokenmacro/impl/ChangesSinceLastBuildMacroTest.java [hunk 1]
```diff
@@ -70,7 +71,8 @@ public void testShouldPrintDate()
 
         String content = changesSinceLastBuildMacro.evaluate(currentBuild, listener, ChangesSinceLastBuildMacro.MACRO_NAME);
 
-        assertEquals("Oct 21, 2013 7:39:00 PM", content);
+        // Java 9 changed the SHORT date format...
```
Comment (oleg-nenashev) [inline src/test/java/org/jenkinsci/plugins/tokenmacro/impl/ChangesSinceLastBuildMacroTest.java]:
```suggestion
        // Java 9 changed the SHORT date format... https://www.oracle.com/technetwork/java/javase/9-relnote-issues-3704069.html#JDK-8008577
```

File: src/test/java/org/jenkinsci/plugins/tokenmacro/impl/ChangesSinceLastSuccessfulBuildMacroTest.java [hunk 2]
```diff
@@ -146,7 +146,9 @@ public void testShouldPrintDate()
 
         String contentStr = content.evaluate(currentBuild, listener, ChangesSinceLastSuccessfulBuildMacro.MACRO_NAME);
 
-        Assert.assertEquals("Changes for Build #41\n" + "Oct 21, 2013 7:39:00 PM\n" + "Changes for Build #42\n" + "Oct 21, 2013 7:39:00 PM\n", contentStr);
+        // Date format changed in Java 9, so we have to accomodate the potential additional comma
```
Comment (oleg-nenashev) [inline src/test/java/org/jenkinsci/plugins/tokenmacro/impl/ChangesSinceLastSuccessfulBuildMacroTest.java]:
```suggestion
        // Date format changed in Java 9, so we have to accomodate the potential additional comma
        // See https://www.oracle.com/technetwork/java/javase/9-relnote-issues-3704069.html#JDK-8008577
```

File: src/test/java/org/jenkinsci/plugins/tokenmacro/impl/ChangesSinceLastUnstableBuildMacroTest.java [hunk 3]
```diff
@@ -146,7 +146,9 @@ public void testShouldPrintDate()
 
         String contentStr = content.evaluate(currentBuild, listener, ChangesSinceLastUnstableBuildMacro.MACRO_NAME);
 
-        Assert.assertEquals("Changes for Build #41\n" + "Oct 21, 2013 7:39:00 PM\n" + "Changes for Build #42\n" + "Oct 21, 2013 7:39:00 PM\n", contentStr);
+        // Date format changed in Java 9, so we have to accomodate the potential additional comma
```
Comment (oleg-nenashev) [inline src/test/java/org/jenkinsci/plugins/tokenmacro/impl/ChangesSinceLastUnstableBuildMacroTest.java]:
```suggestion
        // Date format changed in Java 9, so we have to accomodate the potential additional comma
        // See https://www.oracle.com/technetwork/java/javase/9-relnote-issues-3704069.html#JDK-8008577
```

**Comment by batmat** [comment]
@slide care to review and merge if ok? Thanks!

**Comment by batmat** [comment]
@slide I tested using PCT, I believe this fixes a few issues.
This would be great if you can release for a better Java 11 readiness?

Thanks!

**Comment by slide** [comment]
I can do it today

**Pull request merged**